### ISSUE TITLE
Screen Renders White [Weird Server/Client side error]

### ISSUE THREAD

**Issue by whatsapp2** [DonBruce64/MinecraftTransportSimulator#721]
Hi MTS mod developer, 
I wanna summarize this issue since it might be a bit of a long story, also trying to give information about what we have discovered about this issue: 
Staff members in our server have been trying to solve an issue regarding MTS and/or UNU (since we don't know which one is causing the problem until I talked to Dr. Prof_Luigi and told me that: "...it would be MTS since UNU doesn't actually have any rendering code..."). 

-When we place a car and add some components like seats/wheels/lights our screen renders a solid white, we can still interact with the game, like pressing T and opening chat to /kill the entity that's causing the error.  
-We've been investigating a bit though the latest.log files and when screen is rendered white, console starts spamming: OpenGL error 1281
-Vanilla minecraft entities like the ender dragon or the mushroom cow also renders our screen white. 
-Small models like 

When you get this rendering error and then login off to the minecraft main menu, this is what happens:

![2021-02-22_22 06 54](https://user-images.githubusercontent.com/25994176/108799519-2139b780-7556-11eb-955e-9ed9edc6ee43.png)
(yes this is actually how the main menu looks after getting the error)

This is our client/server modlist:
https://pastebin.com/vD1bsrxd

This is one of our most recent latest.log file we have available:
https://pastebin.com/rmYgynmJ

**Comment by PhilSkinn** [comment]
I'm helping whatsapp2 and it seems to have something to do with Immersive Railroading and Immersive Vehicles not playing well together. If you load a train and then load a Immersive Vehicles vehicle (in this case, from UNU civilian pack), it throws a white screen up. If you only load up an Immersive Vehicles vehicle, and not a train from IR, it works perfectly. Hopefully this helps the issue to be resolved.

**Comment by DonBruce64** [comment]
Very interesting.  I will say one thing: IV does absolutely 0 rendering once you get out of the world.  None of the rendering routines should be called.  And certainly none of the Vanilla routines are ever called by IV.  I say that, because if that's the main menu from the server selection/world selection screen, that's not anything IV is directly doing as there's no code to render hotbars and whatnots.

Now, as to what's being rendered there, it looks like it's a model of some sort with the backface culling all out of whack.  I've occasionally seen that when IV is installed, but I don't know if it's IV that's doing that or not.  IV does have front-face culling, but it disables it in the same routine it enables it in, so I'm not sure how it'd get stuck on like that.  Do you happen to know which model that is?

Also, you  said cows and ender dragons turn white.  Does that mean they just don't have color?  Or does looking at them make the screen white too?

Finally, did IR release a new version recently? I've heard rumors about them implementing a new rendering system, so it might be that which is the source of the issue.  Trying to narrow things down here.

**Comment by PhilSkinn** [comment]
The model that was bugged was the Baron Wagon from UNU version 4.2.0. Whenever we look at the mooshroom cows, the screen would turn white, much like looking at vehicles would. I'll check the earlier Immersive Railroading versions to see if that's the issue. Thank you for your time.

**Comment by whatsapp2** [comment]
"Finally, did IR release a new version recently?" Yes they did, but we have been experiencing this white screen rendering error since IR 1.7. Also another weird thing is that, in out previous iteration, there was no rendering issue whatsoever, we where using IR 1.7.3 and IV 1.12.2-18.1.4 

"Do you happen to know which model that is?" I have no idea what might be the model tbh.

"Also, you said cows and ender dragons turn white. Does that mean they just don't have color? Or does looking at them make the screen white too?" - they render the screen white too.

**Comment by DonBruce64** [comment]
Interesting.  When the cows make the screen turn white, is this with IV installed, or all the time?  If you don't have any IV vehicles around, does the screen still turn white?  I'm struggling to see anything in my code that would affect *cow* rendering routines...

**Comment by whatsapp2** [comment]
When that happens, IV is installed both in the server and client side, we might think it's an error with the model placement, mushroom cows have mushrooms on their back, which might be classified as a rare model for minecraft rendering. Also the ender dragon has some extra models to make it work like the tail, wings, body and head.

Another thing we found is that, motorcycles (or was it the scooters?) from unu doesn't give us a white screen, it might be because it's a small model, less than 1 block

**Comment by PhilSkinn** [comment]
I spoke to the people in the IR server, and they mentioned that they haven't implemented any new rendering system in. I've also done a bit of testing on a private server and the bug isn't able to be replicated. I'll continue to test, but from the looks, it's something to do with IV.

**Comment by PhilSkinn** [comment]
So the issue (for one reason or another) has been resolved. We have no clue how or why this happened, but it's been solved (permanently, hopefully). There were a few other minor bugs that we found, (such as a player being teleported to "Nan,Nan,Nan, and the players arms come off their body), but those are just minor issues that I'm sure will be dealt with. As for right now, we are good, and we thank you for your help in solving the issue. We'll let you know if it happens again.
